core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -142,30 +137,22 @@ private[spark] class ExecutorAllocationManager(
   // Executors that have been requested to be removed but have not been killed yet
   private val executorsPendingToRemove = new mutable.HashSet[String]
```

Comment:
Is it possible also moving this into `ExecutorMonitor` ? So that `ExecutorAllocationMonitor` would be more pure by removing `onExecutorRemoved` and `ExecutorMonitor` would be more knowledgeable about executor's state, including  pending.